ui/src/external/monaco.flux.completions.ts

```diff
@@ -13,6 +13,83 @@ const m2p = new MonacoToProtocolConverter(),
   p2m = new ProtocolToMonacoConverter()
 
 export function registerCompletion(monaco: MonacoType, server: LSPServer) {
+  monaco.languages.registerDocumentSymbolProvider(FLUXLANGID, {
+    provideDocumentSymbols: async (model, _token) => {
+      try {
+        const uri = model.uri.toString()
+        const symbols = await server.symbols(uri)
+
+        return p2m.asSymbolInformations(symbols)
+      } catch (e) {
+        console.error(e)
+      }
+    },
+  })
+
+  monaco.languages.registerFoldingRangeProvider(FLUXLANGID, {
+    provideFoldingRanges: async (model, _context, _token) => {
+      try {
+        const ranges = await server.foldingRanges(model.uri.toString())
+
+        return p2m.asFoldingRanges(ranges)
+      } catch (e) {
+        // console.error(e)
+      }
+
+      return []
+    },
+  })
+
+  monaco.languages.registerDefinitionProvider(FLUXLANGID, {
+    provideDefinition: async (model, position, _token) => {
+      const uri = model.uri.toString()
+      const pos = m2p.asPosition(position.lineNumber, position.column)
+
+      try {
+        const loc = await server.definition(uri, pos)
+
+        return p2m.asLocation(loc)
+      } catch (e) {
+        // console.log(e)
+      }
+
+      return []
+    },
+  })
+
+  monaco.languages.registerReferenceProvider(FLUXLANGID, {
+    provideReferences: async (model, position, context, _token) => {
+      const uri = model.uri.toString()
+      const pos = m2p.asPosition(position.lineNumber, position.column)
+
+      try {
+        const locs = await server.references(uri, pos, context)
+
+        return (locs || []).map(loc => p2m.asLocation(loc))
+      } catch (e) {
+        // console.log(e)
```

Comment:
why keep the commented code?

Reply:
maybe just a TODO/Note comment would work too, just wanna make sure we remember that this try catch is important and don't remove it

Reply:
just moved the default return into the catch